cpg-core/src/main/java/de/fraunhofer/aisec/cpg/GraphTransformation.kt

```diff
@@ -0,0 +1,158 @@
+        block()
+        popFromHandleLog(trackedObject)
+    }
+
+    /**
+     * Executes the provided block of code while holding the tracked object on the stack. This should be used when the
+     * component traverses a hierarchical construct of interest, e.g. parser-AST, CPG-AST, and when the block of code yields
+     * a result that has to be written into a variable.
+     *
+     */
+    fun <T, S> withNodeInLogReturning(trackedObject: S, block: () -> T): T {
+        pushToHandleLog(trackedObject)
+        val ret = block()
+        popFromHandleLog(trackedObject)
+        return ret
+    }
+
+    /**
+     * An extension to <code>replaceNodeInLog<\code> with an extension that tracks another object after the execution of a block
+     */
+    fun <T, S> replaceWithReturningNodeInLog(trackedObject: S, block: () -> T): T {
+        popFromHandleLog(trackedObject)
+        val ret = block()
+        pushToHandleLog(ret)
+        return ret
+    }
+
+    /**
+     * Used to replace the provided tracked object <code>old</code> with a <code>new<\code> one. This should be used when
+     * the component handles nodes in sequence and the prior handled object does not matter for the error tracking. Currently
+     * used as default for non-hierarchical components.
+     */
+    fun <S> replaceNodeInLog(old: S, new: S) {
+        popFromHandleLog(old)
+        pushToHandleLog(new)
+    }
+
+    companion object {
+
+        val githubIssueGuide: String =
+            "\tTo report this Issue visit https://github.com/Fraunhofer-AISEC/cpg/issues/new?&template=bugreport-from-cpg-traces.md&title=%s\n" +
+                "\tIf possible: \n" +
+                "\t\t* paste this message and stack trace for us to locate the issue.\n" +
+                "\t\t* past the parsed code that cause the issue from your source, the location is referenced by the lines 'at processing of ... in ...'\n" +
+                "\t\t* tell us if you used the default passes and language frontends, or made any changes to the TranslationConfiguration, e.g. registered new passes or frontends, or deactivated any."
+
+        open fun getTranslationExceptionWithHandledStack(
+            gt: GraphTransformation,
+            originalException: Exception
+        ): TranslationException {
+            val componentName = gt.javaClass.simpleName
+            val baseErrorName = originalException.javaClass.simpleName
+            val exceptionMessage = "$baseErrorName in $componentName"
+            var customErrorMessage =
+                "$exceptionMessage\n\n${githubIssueGuide.format(URLEncoder.encode(exceptionMessage, "utf-8"))}\n"
+            val size = gt.parserObjectStack.size
+            val stackTrace =
+                Array<StackTraceElement>(size) { i ->
+                    StackTraceElement(
+                        "processing of ${gt.parserObjectStack[size - 1 - i].javaClass.name} located in ${gt.getLocationString(gt.parserObjectStack[size - 1 - i])}",
```

Comment:
The `.javaClass.name` approach does not work for the python framework, as all nodes are of type `jep.python.PyObject`. I assume there is a similar issue with go?
Those frameworks will not be able to make use of the functionality in this PR if we keep it like this.